README.adoc

```diff
@@ -57,6 +57,103 @@ IMPORTANT: Custom Resource Definitions (CRD) are cluster-wide objects and you ne
 operation can be done *once per cluster*. So, if the `kamel install` operation fails, you'll be asked to repeat it when logged as admin.
 For Minishift, this means executing `oc login -u system:admin` then `kamel install --cluster-setup` only for first-time installation.
 
+==== OLM deployment from source
+
+The following steps assume that
+
+- you've already built the camel-k image using `make images` and made it available in the cluster as an imagestream
+- you've already built the bundle image using `make bundle` and have pushed it to some registry
+
+To perform OLM (Operator Lifecycle Manager) based deployment of camel-k, built from source locally on a Openshift cluster, you can follow the steps below.
+
+Login to the cluster using the standard "oc" tool, create new project, do other basic setup. Reference commands below
+
+```
+oc login -u <user> -p <password>
+oc new-project camelk || true
+oc policy add-role-to-group system:image-puller system:serviceaccounts --namespace=camelk || true
+oc patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
+HOST=$(oc get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
+echo $HOST
+podman login -u <user> -p $(oc whoami -t) --tls-verify=false $HOST
+```
+
+Now, we need to build a catalog image (from the previously built bundle image, which has been pushed to some registry) and make it available in the cluster as an imagestream. And to achieve this, we need to build the opm binary and also build the image named `upstream-opm-builder` which is used by opm to build the catalog image. Here are the steps to build the opm binary and the `upstream-opm-builder` image.
+
+```
+mkdir $GOPATH/src/github.com/operator-framework
+cd $GOPATH/src/github.com/operator-framework
+git clone https://github.com/operator-framework/operator-registry.git
+cd operator-registry
+git checkout v1.15.3
+make build
+export PATH=$PATH:`pwd`/bin
+podman build -f upstream-opm-builder.Dockerfile -t quay.io/operator-framework/upstream-opm-builder:latest .
+```
+
+Now, the catalog image can be built and pushed to the imagestream as:
+
+```
+opm index add -u podman --bundles <registry>/<repo>/camel-k-bundle:<tag> --tag $HOST/camelk/camel-k-catalog:<tag> -p podman
+podman push $HOST/camelk/camel-k-catalog:<tag> --tls-verify=false
+```
```

Comment:
It may be useful to mention the option to change the container runtime, for platforms that are not supported by Podman.